server/auth.ts

```diff
@@ -95,6 +121,29 @@ if (process.env.NODE_ENV !== "test") {
       res.send("<script>setInterval(window.close)</script>");
     }
   );
+  
+  router.get("/login-discord", (req, res) => passport.authorize("discord")(req, res));
+  
+  router.get(
+    "/discord/callback",
+    (req, res, next) =>
+      passport.authorize("discord", { failureRedirect: "/login-discord" })(req, res, next),
+    async (req: Request<BaseRequestArgs, BaseRequestArgs>, res) => {
+      if (req.user && req.account) {
+        // Associate discord account with logged-in user
+        await User.findByIdAndUpdate(req.user._id, {
+          $set: {
+            discordId: req.account.id,
+            discord: `${req.account.username}#${req.account.discriminator}`,
```

Comment:
i suppose we'll have to change this since discord is phasing out discriminators. testing locally, it looks like the discriminator field is `"0"` for me.

maybe can check for it like this?
```

function getFullDiscordUsername(account: DiscordAccount): string {
  if (account.discriminator && account.discriminator !== "0") {
    return `${account.username}#${account.discriminator}`;
  }
  return account.username;
}
```